core/vm/evm.go

```diff
@@ -336,18 +337,37 @@ func (c *codeAndHash) Hash() libcommon.Hash {
 func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64, value *uint256.Int, address libcommon.Address, typ OpCode, incrementNonce bool) ([]byte, libcommon.Address, uint64, error) {
 	var ret []byte
 	var err error
+	var gasConsumption uint64
+
+	if evm.config.Debug {
+		if evm.interpreter.Depth() == 0 {
+			evm.config.Tracer.CaptureStart(evm, caller.Address(), address, false /* precompile */, true /* create */, codeAndHash.code, gas, value, nil)
+			defer func() {
```

Comment:
probably redundant new func creation, can do
`defer evm.config.Tracer.CaptureEnd(ret, gasConsumption, err)`

Reply:
@AskAlexSharov when I did that, the err was captured at the point of the call of the defer, rather than when the defer is called (therefore the error was passed as nil).